usermods/audioreactive/audio_reactive.h

```diff
@@ -638,13 +645,11 @@ class AudioReactive : public Usermod {
 #ifdef ARDUINO_ARCH_ESP32
     // used for AGC
     int      last_soundAgc = -1;   // used to detect AGC mode change (for resetting AGC internal error buffers)
-    double   control_integrated = 0.0;   // persistent across calls to agcAvg(); "integrator control" = accumulated error
-
-
+    float    control_integrated = 0.0f;   // persistent across calls to agcAvg(); "integrator control" = accumulated error
     // variables used by getSample() and agcAvg()
     int16_t  micIn = 0;           // Current sample starts with negative values and large values, which is why it's 16 bit signed
-    double   sampleMax = 0.0;     // Max sample over a few seconds. Needed for AGC controller.
-    double   micLev = 0.0;        // Used to convert returned value to have '0' as minimum. A leveller
+    float    sampleMax = 0.0f;    // Max sample over a few seconds. Needed for AGC controller.
+    float    micLev = 0.0f;       // Used to convert returned value to have '0' as minimum. A leveller
```

Comment:
Just a note for understanding - neither `getSample()`  (runs microphone input filters) nor `agcAvg()` (runs AGC algo) are time critical - they are just a handful of math expressions in a sequence, without loops. So it's ok to have some computations in double, because even -C3 will be able to handle them without problem.
Only `FFTcode()` is time critical. 

In case your intention was to avoid linking `double` libm functions, there are other libs (like ArduinoFFT) that still use double, so I doubt this will help much.

Reply:
ArduinoFFT uses float when you construct FFT object using float in template.
It will use double if you want SQRT approximation (which we don't).

Reply:
90% agreed. Only the windowing function (which we need) are still using sin/cos instead of sinf/cosf

https://github.com/kosme/arduinoFFT/blob/0da88512f90b0c8c6c5d50127045ee4f745e343b/src/arduinoFFT.cpp#L379-L382

Reply:
AFAIK C++ will use correct overloads.
https://en.cppreference.com/w/cpp/numeric/math/cos

Reply:
> And the library is still full of "double" constants (1.0, 2.0, etc)

Some are stored in `float[]`so they are truncated during compilation. But the comment is valid for in-line computations.
We should open an issue with @kosme perhaps.